vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -17,29 +17,42 @@ def initialize(occupancy)
       def process
         return unless occupancy.account_id?
 
-        ActiveRecord::Base.transaction do
-          create_order
-          create_order_detail
-
-          order.validate_order!
-          order.purchase!
-        end
+        find_or_create_order
+        fulfill_order if occupancy.complete?
```

Comment:
I'm not fully sure of what's going on behind the scenes of `fulfill_order`, but it seems like we'd want to hang onto the order--if we just created the order and the fulfill fails, I think we still want the order around